src/aggregation/metric/stats.rs

```diff
@@ -143,12 +300,266 @@ impl IntermediateStats {
+                self.delta_sum_for_squares_elastic = other.delta_sum_for_squares_elastic
+            } else {
+                if other.intermediate_stats.count == 1 {
+                    self.collect(other.intermediate_stats.sum);
+                } else if self.intermediate_stats.count == 1 {
+                    let sum = self.intermediate_stats.sum;
+                    self.intermediate_stats = other.intermediate_stats;
+                    self.sum_of_squares = other.sum_of_squares;
+                    self.sum_of_squares_elastic = other.sum_of_squares_elastic;
+                    self.mean = other.mean;
+                    self.delta_sum_for_squares_elastic = other.delta_sum_for_squares_elastic;
+                    self.collect(sum);
+                } else {
+                    // parallel version of Welford's online algorithm
+                    // the mean is computed using sum and count because
+                    // it's more precise (and sum is already available)
+                    let new_count = self.intermediate_stats.count + other.intermediate_stats.count;
+                    let delta = other.mean - self.mean;
+                    self.sum_of_squares += other.sum_of_squares
+                        + delta
+                            * delta
+                            * self.intermediate_stats.count as f64
+                            * other.intermediate_stats.count as f64
+                            / new_count as f64;
+                    self.intermediate_stats.count = new_count;
+                    // self.mean=self.mean + delta*other.count as f64/new_count as f64;
+                    self.mean = (self.intermediate_stats.sum as f64
+                        + other.intermediate_stats.sum as f64)
+                        / new_count as f64;
+                    self.intermediate_stats.sum += other.intermediate_stats.sum;
+                    self.intermediate_stats.delta += other.intermediate_stats.delta;
+                    self.sum_of_squares_elastic += other.sum_of_squares_elastic;
+                    self.delta_sum_for_squares_elastic += other.delta_sum_for_squares_elastic
+                }
+            }
+        }
+    }
+
+    /// Computes the final stats value.
+    pub fn finalize(&self) -> Box<ExtendedStats> {
+        let min = if self.intermediate_stats.count == 0 {
+            None
+        } else {
+            Some(self.intermediate_stats.min)
+        };
+        let max = if self.intermediate_stats.count == 0 {
+            None
+        } else {
+            Some(self.intermediate_stats.max)
+        };
+        let avg = if self.intermediate_stats.count == 0 {
+            None
+        } else {
+            Some(self.mean)
+        };
+        let sum_of_squares = if self.intermediate_stats.count == 0 {
+            None
+        } else {
+            Some(self.sum_of_squares_elastic)
+        };
+        let variance = if self.intermediate_stats.count <= 1 {
```

Comment:
all these different but same ifs are error prone and should be collapsed into one condition

Reply:
Ok, my code was inspired by the IntermediateStats one 

https://github.com/quickwit-oss/tantivy/blob/aa26ff50298e8a6a590faf9a66bdd7d2a684ec97/src/aggregation/metric/stats.rs#L119-L133

one should I refactor that one as well?

Reply:
This is also not nice, but there are all the same check so it's easy to proofread